include/grpc/impl/codegen/port_platform.h

```diff
@@ -518,4 +518,18 @@ typedef unsigned __int64 uint64_t;
 #define __STDC_FORMAT_MACROS
 #endif
 
+#if defined(__clang__) && defined(__has_cpp_attribute) \
+    && !defined(GOOGLE_PROTOBUF_OS_APPLE)
+# if defined(GOOGLE_PROTOBUF_OS_NACL) || defined(EMSCRIPTEN) || \
+     __has_cpp_attribute(clang::fallthrough)
+#  define GOOGLE_FALLTHROUGH_INTENDED [[clang::fallthrough]]
```

Comment:
naming: this probably should be called `GOOGLE_..`.   `GRPC_FALLTHROUGH_INTENDED` or `GPR_FALLTHROUGH_INTENDED`?